### ISSUE TITLE
Fix Transformers models relying on tensorflow and jax

### ISSUE THREAD

**Issue by RobinPicard** [dottxt-ai/outlines#1692]
I recently tried to run a model using `FlaxAutoModelForCausalLM` and generation ran into an error. Our testing for transformers models using `jax` and `tensorflow` for logits handling is currently weak as we only check whether we can initialize the model. We should fix errors and improve testing,

**Comment by RobinPicard** [comment]
Ah that's good news for us. Closing the issue then.
